### DISCUSSION TITLE
Improving filtering for string script?

### DISCUSSION THREAD

**Discussion by mlesniak** [babashka/babashka#1246] [Q&A] [answered]
Hi,

I'm currently learning Clojure and want to use it for shell scripts first. As a playground example I'd like to write a small (contrived) filter script which filters each line for a string, e.g. "root":

```clojure
#!/usr/bin/env bb -i

;; Uses [Babashka](https://babashka.org/)
;; brew install babashka

(defn contains [string] (fn [y] (.contains y string)))
(doseq [v (filter (contains "root") *input*))]
  (println v))
```

Honestly it seems to be way too complicated for such a simple task but I can't see how I can improve (shorten) this snippets, hence any advice is appreciated.

Best,
Michael

**Comment by borkdude** [comment, marked as answer]
@mlesniak Clojure offers the `clojure.string` namespace for this which has a function `includes?`:

``` clojure
(require '[clojure.string :as str])
(->> (slurp *in*)
     (str/split-lines)
     (filter #(str/includes? % "root"))
     (run! println))
```

**Reply by mlesniak** [reply]
Thanks @borkdude, that was very helpful!

**Comment by borkdude** [comment]
> Honestly it seems to be way too complicated for such a simple task

Babashka shines mostly when you need more than 5 lines of bash... that's around the limit when bash starts to bite me ;).

**Reply by mlesniak** [reply]
I completely agree -- right now I try to write even smaller scripts via `bb` to improve learning ;-).
